components/dash-core-components/src/components/Geolocation.react.js

```diff
@@ -0,0 +1,193 @@
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * The CurrentLocation component gets geolocation of the device from the web browser.  See more info here:
+ * https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API
+ */
+
+export default class Geolocation extends Component {
+    constructor(props) {
+        super(props);
+        this.success = this.success.bind(this);
+        this.error = this.error.bind(this);
+        this.updatePosition = this.updatePosition.bind(this);
+    }
+    updatePosition() {
+        if (!navigator.geolocation) {
+            this.error({
+                code: 999,
+                message: 'Your browser does not support Geolocation',
+            });
+            if (this.props.show_alert) {
+                alert(`ERROR(${this.error.code}): ${this.error.message}`);
+            }
+        } else {
+            this.props.setProps({
+                update_now: false,
+            });
```

Comment:
Would it be better to put this at the very beginning of `updatePosition`, so it's cleared even if `geolocation` is unsupported? Also maybe wrap it in `if (this.props.update_now) { ... }` so we don't call `setProps` unnecessarily?